### PR TITLE
Fix typos, modernize and simplify some code

### PR DESCRIPTION
**Pull request by eumiro**
A few independent commits for the tox4 rewrite in five commits:

1. fix some typos
2. push minimal required Python version from 3.5 to 3.6
3. simplify code to parse string into smaller parts (here the flake8 CI checker dislikes `str.rpartition` and thinks it's a typo)
4. simplify a nested condition
5. remove one unused variable

### CODE REVIEW

File: src/tox/config/loader/api.py [hunk 1]
```diff
@@ -16,17 +16,13 @@
...
-        if split_at == -1:
+        key, equal, self.value = value.partition("=")
+        if not equal:
             raise ArgumentTypeError(f"override {value} has no = sign in it")
-        key = value[:split_at]
-        ns_at = key.find(".")
-        self.namespace = "" if ns_at == -1 else key[:ns_at]
-        self.key = key[ns_at + 1 :]
-        self.value = value[split_at + 1 :]
+        self.namespace, _, self.key = key.rpartition(".")
```
Comment (gaborbernat) [inline src/tox/config/loader/api.py:22]: you need to add this to whitelist.txt at root

File: src/tox/config/loader/api.py [hunk 2]
```diff
@@ -16,17 +16,13 @@
...
 
     def __repr__(self) -> str:
-        return f"{self.__class__.__name__}('{self}')"
+        return f"{self.__class__.__name__}({self!r})"
```
Comment (gaborbernat) [inline src/tox/config/loader/api.py:25]: revert this, this generates an infinite stack, you're essentially calling yourself

File: src/tox/config/loader/ini/factor.py [hunk 3]
```diff
@@ -19,11 +19,7 @@ def filter_for_env(value: str, name: Optional[str]) -> str:
                 overall.append(content)
         else:
             for group in factors:
-                for a_name, negate in group:
-                    contains = a_name in current
-                    if not ((contains is True and negate is False) or (contains is False and negate is True)):
-                        break
-                else:
+                if all((a_name in current) ^ negate for a_name, negate in group):
```
Comment (gaborbernat) [inline src/tox/config/loader/ini/factor.py:22]: This is one of those now it's shorter but harder to read. I mean I'm not against keeping this, but just looking at it I'm not 100% it's correct, though I trust the test suite.

File: src/tox/util/pep517/backend.py [hunk 4]
```diff
@@ -23,11 +23,11 @@ def __init__(self, backend_module, backend_obj):
     def __call__(self, name, *args, **kwargs):
```
Comment (gaborbernat) [inline src/tox/util/pep517/backend.py:23]: Revert changes on this file. This needs to be python 2 compatible to allow python 2 backends. See https://github.com/eumiro/tox/blob/fix-typos/.pre-commit-config.yaml#L20-L21 pinning it so.

**Review by gaborbernat — CHANGES_REQUESTED** [review]
Most are good, some changes need to be reverted.

File: src/tox/config/loader/api.py (hunk 1 above)
Comment (eumiro) [reply src/tox/config/loader/api.py:22]: Okay, thanks.

File: src/tox/config/loader/api.py (hunk 2 above)
Comment (eumiro) [reply src/tox/config/loader/api.py:25]: Oh, that was a bad idea. Reverted, thank you.

File: src/tox/config/loader/ini/factor.py (hunk 3 above)
Comment (eumiro) [reply src/tox/config/loader/ini/factor.py:22]: Please decide what fits you. From my point of view, my proposal is slightly slower to read per word but more compact and therefore needs less time and mental energy to understand. But that's my point, so please feel free to decide.

File: src/tox/util/pep517/backend.py (hunk 4 above)
Comment (eumiro) [reply src/tox/util/pep517/backend.py:23]: I was actually wondering why there was so much `.format` code left. I'll drop this commit.

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**